packages/schematics/angular/guard/index.ts

```diff
@@ -7,14 +7,35 @@
  */
 
 import { Rule, SchematicsException } from '@angular-devkit/schematics';
+
 import { generateFromFiles } from '../utility/generate-from-files';
+
 import { Implement as GuardInterface, Schema as GuardOptions } from './schema';
 
 export default function (options: GuardOptions): Rule {
-  if (!options.implements) {
-    throw new SchematicsException('Option "implements" is required.');
+  if (!options.implements && !options.guardType) {
+    throw new SchematicsException('Option "implements" or "guardType" is required.');
+  }
+  if (options.implements && options.implements.length > 0 && options.guardType) {
+    throw new SchematicsException('Options "implements" and "guardType" cannot be used together.');
+  }
+
+  return options.guardType ? functionalGuardRule(options) : classBasedGuardRule(options);
+}
+
+function functionalGuardRule(options: GuardOptions): Rule {
+  if (!options.guardType) {
+    throw new SchematicsException('Option "implements" or "guardType" is required.');
   }
+  const guardType = options.guardType.replace(/^can/, 'Can') + 'Fn';
```

Comment:
Out of curiosity why not use set the value of the enum as `Can**Fn` directly?

Reply:
I wanted to map the names to the `Route` properties because it's less typing and less to remember.

Reply:
My unsolicited 2 cents:
We discussed this for our book recently and differentiated between the *type of the guard* and the *TypeScript type of the function*. A "CanActivate guard" is the concept while `CanActivateFn` is the TS type for it (or even `CanActivate` for classes).

I think this distinction between concept and technical type is important. `guardType` describes the concept.

tl;dr; Agree with @atscott :smile: